migrations.go

```diff
@@ -33,20 +28,46 @@ func (m *Migration) String() string {
 	return strconv.FormatInt(m.Version, 10)
 }
 
+// A Group is a collection of migrations supporting registration and execution.
+type Group struct {
+	// DiscoverSQL enables auto-discovery and registration of SQL files as migration steps.
+	DiscoverSQL bool
+	// TableName is the name of the table created to track applied migrations.
+	TableName string
```

Comment:
Making TableName part of the group avoids different migrations fighting over the same global state controlling the table name.

Reply:
Can we instead add `Group.SetTableTame(string)` and `Group.AutoDiscoverSQL(bool)`? This way it is easier to change API later.

Reply:
Exposing methods here rather than struct fields directly definitely makes sense. I've taken it a little bit further actually and only exposed `Collection` as an interface rather than as a struct with hidden fields.

I opted for `WithTableName(string)` and `WithAutoDiscoverSQL(bool)` though, to better support a single-statement initialization of a package-wide `Collection` like:

```
var MigrationCollection = migrations.NewCollection().
  WithTableName("my_table").
  WithAutoDiscoverSQL(true)
```